server/core/region_tree_test.go

```diff
@@ -321,7 +321,7 @@ func (s *testRegionSuite) TestRandomRegionDiscontinuous(c *C) {
 
 func checkRandomRegion(c *C, tree *regionTree, regions []*RegionInfo, startKey, endKey []byte) {
 	keys := make(map[string]struct{})
-	for i := 1; i < 20; i++ {
+	for i := 1; i < 50; i++ {
```

Comment:
why can it reduce the possibility of failure by enlarging the loop count ?
IMO, it will increase because we will check assert in every loop.

Reply:
The purpose of this loop is to get all regions as possible as it meets the requirement. But due to the random, we may not get all. Enlarge the loop count will reduce the possibility of failure.

Reply:
oh, so it enlarge coverage to increase test validation rather than test pass rate?

Reply:
I think it will be ok to simply random get some regions then verify they all belong to the range.

Reply:
> oh, so it enlarge coverage to increase test validation rather than test pass rate?

@lhy1024 Yes.

> I think it will be ok to simply random get some regions then verify they all belong to the range.

@disksing The original thought about this test is that for a given range, we can randomly choose to pick a region and **all regions within this range can be picked**.

Reply:
I knew. But there are always fail probability if you want to test it.

Reply:
Right. I think after we enlarge the loop count, the probability can be omitted. I have run 1000 times, and there is no failure.

Reply:
how about `for i := 1; i < 10000 && len(keys) < len(region); i++`?

Reply:
It can verify that every region can be accessed and can exit the loop earlier.

Reply:
why to use `len(keys) < len(region)`? It seem to be conflict with `c.Assert(keys, HasLen, len(regions))` in 336.

Reply:
Not really. If there is a bug that some region cannot be accessed, the loop will stop after 10000 iterations and the assert will fail.

Reply:
Make sense, changed.